src/datastructure/lcm_hashtable.h

```diff
@@ -114,24 +115,23 @@ static inline int LCM_hashtable_insert(LCM_hashtable_t* h,
     if (tag == cmp_key) {
       *value = hentry->entry.val;
       hentry->entry.tag = LCM_HASHTABLE_EMPTY;
-      LCIU_release_spinlock(&master->control.lock);
-      LCM_DBG_Log(LCM_LOG_DEBUG, "hashtable",
-                  "insert (%lx, %p, %d), return 0\n", key, value, type);
-      return 0;
+      found = true;
+      break;
     } else if (tag == LCM_HASHTABLE_EMPTY) {
-      // Ortherwise, if the tag is empty, we record the slot.
+      // Otherwise, if the tag is empty, we record the slot.
       // We can't return until we go over all entries.
       if (empty_hentry == NULL) empty_hentry = hentry;
     } else {
       // If we are still seeing some non-empty,
       // push that empty entry even further.
-      empty_hentry = NULL;
+      // FIXME: Why pushing the empty entry further?
+      //      empty_hentry = NULL;
     }
 
     hentry++;
     checked_slot++;
     // If we go over all entry, means no empty slot.
-    if (checked_slot == (LCM_HASHTABLE_WIDTH - 1)) {
+    if (checked_slot % (LCM_HASHTABLE_WIDTH - 1) == 0) {
```

Comment:
Why are you using modulo rather than the prior reset `checked_slot` code? Integer division by non-power-of-two (which `LCM_HASHTABLE_WIDTH - 1` is) should be a lot more expensive, no?

Reply:
Oh I see you're using it later on for perf counters.